### DISCUSSION TITLE
Anacrusis and "Reverse anacrusis"?

### DISCUSSION THREAD

**Discussion by FakenameMusic** [helgoboss/helgobox#1145] [Q&A]
There's 2 things I'm trying to achieve, and fiddling with the  Synchronisation / Dynamic / Static controls for the cell is not getting me there.

The first is a simple anacrusis.

The second is like a sort of reverse anacrusis. An anacrusis - but one that takes place at the end of the first bar when a new row/section is triggered. (I've searched but cannot find an agreed musical term for this - but think of it like a melody that starts late in the bar)

It's essentially a cell state that says "**1)** When you first play this cell, wait until the 3rd beat of the 1st bar to output audio. **2)** Start playing the audio from the 3rd beat in the file. **3)** Then continue to loop back to the start and play in full as normal for each subsequent loop afterwards"

<img width="1126" alt="Screenshot 2024-08-29 at 15 36 29 jpeg" src="https://github.com/user-attachments/assets/0d30e61f-2d77-49eb-8cb4-a6d1c16d09e2">

In the example above, when I change from the Verse to the Chorus I want to trigger the chorus row at the green "x" and have the Purple "Melody" cell wait until the third beat of the bar to play, and start the audio playback at the third beat of the loop that it's playing. This would avoid the notes at the end of the phrase being heard *first* rather than at the end of the phrase as they should be.

Just to make this clearer - the Purple part of the "Wav" I have drawn for the Melody should be heard first, and the yellow part second (I have drawn it in lighter in the first bar of the chorus where it should not be heard until the purple bar begins.

**Comment by helgoboss** [comment]
The reverse anacrusis thing sounds a bit to me like it's a normal anacrusis, just triggered one bar later?

You can do a anacrusis (pickup beat) in Playtime 2 by setting the "Downbeat" of the clip to 1 beat. In order to hear the anacrusis, you need to trigger clip playback early enough (more than 1 beat before start of the bar).

**Comment by FakenameMusic** [comment]
>_The reverse anacrusis thing sounds a bit to me like it's a normal anacrusis, just triggered one bar later?_

That's one way of thinking of it. But in practical / live performance that means that you can't trigger the chorus of a song by hitting the next row on your controller - you have to hit it once to trigger the chorus - wait for the chorus to start and then trigger it again. So in a live performance setting if you wanted to trigger a chorus and then, say, play a synth part as the chorus starts, or switch to your guitar for the chorus, you couldn't, because you are still required to manually start the cell with the anacrusis.

I can think of work arounds where you add a one-shot MIDI cell to your chorus that triggers the cell with a trigger for the anacrusis cell  after the start of the 1st bar of the chorus, but it seems less elegant, like a work-around,  and more work for the user. But I understand where you are coming from.

There's 3 different controls for adjusting start point in the loop and the length in different ways, so why not one for where in the bar that you start in the media item? Just a suggestion :)

>_You can do a anacrusis (pickup beat) in Playtime 2 by setting the "Downbeat" of the clip to 1 beat. In order to hear the anacrusis, you need to trigger clip playback early enough (more than 1 beat before start of the bar)._

For me the expected behaviour would be that I can take a loop from my song cut in position. If the phrase begins with an anacrusis note on the 3rd beat, I could set my anacrusis for 2 beats and then the audio would play from the start of the third beat. So for example:

The phrase "1-2-3-4-" spoken in time with the 4 quater notes in the bar is imported to Playtime from my project.
I decide I want an anacrusis of "3-4-" which is two beats. . After that the loop will continue "1-2-3-4-" in time with the music. I set the Downbeat to "2 beats"

So in my mind I should get ".......3-4-|1-2-3-4-|1-2-3-4-...etc."

Currently it's a bit counter intuitive - I have to prepare my music with the beat order "3-4-1-2-" before I import it into Playtime, and set the downbeat for 2 - which feels like a work around rather than a function.

Just suggestions anyway  :)

**Reply by helgoboss** [reply]
> > _The reverse anacrusis thing sounds a bit to me like it's a normal anacrusis, just triggered one bar later?_
> 
> That's one way of thinking of it. But in practical / live performance that means that you can't trigger the chorus of a song by hitting the next row on your controller - you have to hit it once to trigger the chorus - wait for the chorus to start and then trigger it again. So in a live performance setting if you wanted to trigger a chorus and then, say, play a synth part as the chorus starts, or switch to your guitar for the chorus, you couldn't, because you are still required to manually start the cell with the anacrusis.
> 
> I can think of work arounds where you add a one-shot MIDI cell to your chorus that triggers the cell with a trigger for the anacrusis cell after the start of the 1st bar of the chorus, but it seems less elegant, like a work-around, and more work for the user. But I understand where you are coming from.
> 
> There's 3 different controls for adjusting start point in the loop and the length in different ways, so why not one for where in the bar that you start in the media item? Just a suggestion :)

Sorry, a bit low on time, so I haven't read that in detail yet. But a few tips:

1. Would be it possible to achieve that by pre-cutting the audio file in REAPER? (I assume you already thought about this, so probably not)
2. Could it be achieved using this feature? https://github.com/helgoboss/helgobox/discussions/1049
3. If not, could you have a look if and how this can be achieved in one of the mainstream clip launchers (Live, Bitwig)? This could serve as inspiration.


> > _You can do a anacrusis (pickup beat) in Playtime 2 by setting the "Downbeat" of the clip to 1 beat. In order to hear the anacrusis, you need to trigger clip playback early enough (more than 1 beat before start of the bar)._
> 
> For me the expected behaviour would be that I can take a loop from my song cut in position. If the phrase begins with an anacrusis note on the 3rd beat, I could set my anacrusis for 2 beats and then the audio would play from the start of the third beat. So for example:
> 
> The phrase "1-2-3-4-" spoken in time with the 4 quater notes in the bar is imported to Playtime from my project. I decide I want an anacrusis of "3-4-" which is two beats. . After that the loop will continue "1-2-3-4-" in time with the music. I set the Downbeat to "2 beats"
> 
> So in my mind I should get ".......3-4-|1-2-3-4-|1-2-3-4-...etc."
> 
> Currently it's a bit counter intuitive - I have to prepare my music with the beat order "3-4-1-2-" before I import it into Playtime, and set the downbeat for 2 - which feels like a work around rather than a function.
> 
> Just suggestions anyway :)

I know what you mean, but that's something different / in addition. You are looking for https://github.com/helgoboss/helgobox/discussions/1001.

The "Downbeat" as it is now is closer to the classical idea of an anacrusis than what you are describing. When you look at a score with an anacrusis, the anacrusis is basically the start of the score. And this is exactly the way how it's modelled in Playtime currently. When you record MIDI, there's even the possibility to record the anacrusis. It will take take the first few notes that you record **before** start of the bar and define it as anacrusis. I might implement something similar for audio in the future. Playtime's stance is "Anacrusis comes first, also in the recording".

If I implement the rotation FR one day (#1001), you will be able to achieve what you want by rotating the contents ("1-2-3-4" => "3-4-1-2") and then change the downbeat from 0.00 beats to 1.00 beats.

**Comment by FakenameMusic** [comment]
OK let me go through all that you have said again when I get a chance this evening.....

But for now, regarding finding out how Ableton Live deals with this: I am asking various people (in real life and online) how Ableton Live deals with this, so some information for you:

I will get back to you on this in more detail when I have more information, but for now:
Part of how Ableton deals with this is that each clip has the option of having definable loop points within it. So you could have, for example, a 6 beat clip where beats 3,4,5,6 are defined as a loop. So the clip plays as 1,2,3,4,5,6-_3,4,5,6_,-_3,4,5,6_...etc. My understanding is that you set the loop points in the clip view GUI.

So for the "reverse anacrusis" step 1 would be to make a longer 2 bar clip with silence at the start, like this (where X = silence and "|" is a bar line):
This is a two bar phrase:
|X-X-3-4-|1-2-3-4|

Step 2: After you import the clip to Live Session view, you would then define the looped part within the clip as the second bar of the two bar clip.

When triggered, It plays through the first bar the clip (including the gap of silence) and then keeps looping the second bar (the defined loop section) **until the next user input**.

What I've yet to get an answer on for this is how you then exit in perfect time on the last repeat - i.e. without having the first two notes of the musical phrase play again a last time (which is not what you want as the musical phrase should have ended), and then the phrase is suddenly cut off as the next section of the song begins at the next bar. 

I'm still collecting more information so I'll get back to you when I have more, as I think there may well be other ways users are dealing with this in Ableton Live.

I will say though, from a user perspective it still seems to me that it would be simpler to take the phrase out, in situ, from your Reaper session, and then define 1 point within it (with a visual interface just like your "static section layout") where you define which beat in the clip should be the start / end point of the looped phrase. It would be very clear and simple, and would require minimal amount of making longer clips, setting loop points, trying to catch the right out point while performing etc.

But anyway, I will try to gather more information and get back to you.

Thanks!

**Comment by FakenameMusic** [comment]
Some people are achieving the "reverse anacrusis" with a feature for clips in Ableton called "Follow Actions" - they are instructions of various kinds that are triggered at the end of a cell playing - which includes triggering other clips and scenes.

When the Chorus scene is triggered the clip with silence at the start (the reverse anacrusis melody) is triggered too, and when it finishes playing it triggers the clip that has the full loop of a bar. Again it's not clear to me yet how you exit this phrase accurately - but certainly there'd at least be a workaround to have a clip in your song setup that triggers a relevant midi note to shut off the full bar loop clip after precisely two beats. 

Broad outline of Folloow Action functions....
https://help.ableton.com/hc/en-us/articles/360019101360-Updates-to-Follow-Actions-in-Live-11

**Reply by helgoboss** [reply]
If you want, vote for #1044. And probably https://github.com/helgoboss/helgobox/discussions/1049?

**Comment by FakenameMusic** [comment]
Excellent. Thanks, I just did 🤘
